src/Runner.Worker/ContainerOperationProvider.cs

```diff
@@ -523,5 +485,74 @@ private void UpdateRegistryAuthForGitHubToken(IExecutionContext executionContext
                 container.RegistryAuthPassword = executionContext.GetGitHubContext("token");
             }
         }
+
+        private async Task AssertCompatibleOS(IExecutionContext executionContext)
+        {
+                   // Check whether we are inside a container.
+            // Our container feature requires to map working directory from host to the container.
+            // If we are already inside a container, we will not able to find out the real working direcotry path on the host.
+#if OS_WINDOWS
+#pragma warning disable CA1416
+            // service CExecSvc is Container Execution Agent.
```

Comment:
all of this is unchanged just moved right?

Reply:
Correct 👍